home-assistant-plugin/custom_components/xknx/__init__.py

```diff
@@ -341,4 +341,13 @@ def async_register(self):
         """Handle entity change."""
         if new_state is None:
             return
-        await self.device.set(float(new_state.state))
+        if new_state.state == "unknown":
+            return
+
+        if self.type == 'binary':
+            if new_state.state == "on":
+                await self.device.set(True)
+            elif new_state.state == "off":
+                await self.device.set(False)
+        else:
+            await self.device.set(new_state.state)
```

Comment:
don't convert every new value to float. DPTArray subclasses expecting `int` didn't work. This way `string` is possible too.